packages/manager/apps/web-office/src/pages/dashboard/users/ModalEditUsers.component.tsx

```diff
@@ -0,0 +1,258 @@
+import queryClient from '@/queryClient';
+import { EDIT_USERS_FORM_SCHEMA } from '@/utils/formSchemas.utils';
+import {
+  getOfficeLicenseQueryKey,
+  putOfficeLicenseDetails,
+} from '@/api/license';
+import { getOfficeUsersQueryKey, putOfficeUserDetail } from '@/api/users';
+import { UserParamsType } from '@/api/api.type';
+
+export default function ModalEditUsers() {
+  const { t } = useTranslation(['dashboard/users/edit', 'common']);
+  const navigate = useNavigate();
+
+  const { serviceName: selectedServiceName } = useParams();
+  const [searchParams] = useSearchParams();
+  const activationEmail = searchParams.get('activationEmail');
+  const licencePrepaidName = searchParams.get('licencePrepaidName');
+
+  const { addError, addSuccess } = useNotifications();
+
+  const goBackUrl = useGenerateUrl('..', 'path');
+  const onClose = () => navigate(goBackUrl);
+
+  const {
+    data: userDetail,
+    isLoading: isUserDetailLoading,
+  } = useOfficeUserDetail(activationEmail, licencePrepaidName);
+
+  const [isLoading, setIsLoading] = useState(
+    !userDetail || isUserDetailLoading,
+  );
+
+  const {
+    control,
+    setValue,
+    getValues,
+    formState: { isDirty, isValid, errors },
+  } = useForm({
+    defaultValues: {
+      firstname: userDetail?.firstName || '',
+      lastname: userDetail?.lastName || '',
+      login: userDetail?.activationEmail.split('@')[0] || '',
+    },
+    mode: 'onChange',
+    resolver: zodResolver(EDIT_USERS_FORM_SCHEMA),
+  });
+
+  useEffect(() => {
+    if (userDetail && !isUserDetailLoading) {
+      setValue('firstname', userDetail.firstName);
+      setValue('lastname', userDetail.lastName);
+      setValue('login', userDetail.activationEmail.split('@')[0]);
+      setIsLoading(false);
+    }
+  }, [userDetail, isUserDetailLoading]);
+
+  const { mutate: editUser, isPending: isSending } = useMutation({
+    mutationFn: (params: UserParamsType) => {
+      return licencePrepaidName
+        ? putOfficeLicenseDetails(licencePrepaidName, params)
+        : putOfficeUserDetail(selectedServiceName, activationEmail, params);
+    },
+    onSuccess: () => {
+      addSuccess(
+        <OdsText preset={ODS_TEXT_PRESET.paragraph}>
+          {t('dashboard_users_edit_message_success')}
+        </OdsText>,
+        true,
+      );
+    },
+    onError: (error: ApiError) => {
+      addError(
+        <OdsText preset={ODS_TEXT_PRESET.paragraph}>
+          {t('dashboard_users_edit_message_error', {
+            error: error.response?.data?.message,
+          })}
+        </OdsText>,
+        true,
+      );
+    },
+    onSettled: () => {
+      queryClient.invalidateQueries({
+        queryKey: [
+          licencePrepaidName
+            ? getOfficeLicenseQueryKey(selectedServiceName)
+            : getOfficeUsersQueryKey(selectedServiceName),
+        ],
+      });
+      onClose();
+    },
+  });
+
+  const handleSaveClick = () => {
+    const { firstname, lastname, login } = getValues();
```

Comment:
I would recommend handling the submit using handleSubmit like this

```
export default function App() {
  const { register, handleSubmit } = useForm<FormValues>()
  const onSubmit: SubmitHandler<FormValues> = (data) => console.log(data)

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input {...register("firstName")} />
      <input {...register("lastName")} />
      <input type="email" {...register("email")} />

      <input type="submit" />
    </form>
  )
}
```

Reply:
It's fixed